### PR TITLE
Jczerk/intellij updates

### PR DESCRIPTION
**Pull request by jmczerk**
Enhanced the IntelliJ dev guide to include some info on memory debugging functionality and code coverage.

### CODE REVIEW

File: docs/dev-guides/intellij.md [hunk 1]
```diff
@@ -221,6 +259,32 @@ methods, properties, inner classes, template parameters, parents, and children.
...
+Running with coverage again prompts as to whether to add the newly collected
+coverage results to the existing results, or replace:
+
+![add_or_replace](./images/intellij/add_or_replace.png)
+
+In this case, the add option was chosen, and the coverage has moved up to 100%
+line coverage for the unit under test:
+
+![full_coverage](./images/intellij/full_coverage.png)
+
```
Comment (jaycarlton) [inline docs/dev-guides/intellij.md:287]: Can you delete these empty sections I left please?

File: docs/dev-guides/intellij.md [hunk 2]
```diff
@@ -213,6 +213,44 @@ It can be launched from anywhere, but if an expression in code is highlighted, t
...
+This in turn brings up a window showing all of the class instances that hold
+references to the object in question:
+
+![references](./images/intellij/references.png)
+
+In this case, the discovery of the presence of the `MXBeanSupport`
+reference led to the discovery that unexpected references were being held by
+[JMX](https://en.wikipedia.org/wiki/Java_Management_Extensions) tracking objects
+(see defect[PF-485](https://broadworkbench.atlassian.net/browse/PF-485) for more
```
Comment (jaycarlton) [inline docs/dev-guides/intellij.md]: nit: since this is a public-facing doc, I tend to try to keep links to private resources like Jira out of it.

**Review by jaycarlton — APPROVED** [review]
Thanks!

**Review by gjuggler — APPROVED** [review]
Thanks John! Apologies for the delay; for dev-guides like this, feel free to move ahead with one reviewer or pull someone else in if I'm the long pole.

**Pull request merged**